drivers/unix/os_unix.cpp

```diff
@@ -890,11 +894,27 @@ Error OS_Unix::open_dynamic_library(const String &p_path, void *&p_library_handl
 		path = get_executable_path().get_base_dir().path_join("../lib").path_join(p_path.get_file());
 	}
 
+#if !defined(__ANDROID_API__) && defined(__linux__) || defined(__FreeBSD__) || defined(__NetBSD__)
+	// If dlinfo is available try to dlopen dynamic library without preceding path
+	if (!FileAccess::exists(path)) {
+		path = p_path.get_file();
+	}
+#else
 	ERR_FAIL_COND_V(!FileAccess::exists(path), ERR_FILE_NOT_FOUND);
+#endif
 
 	p_library_handle = dlopen(path.utf8().get_data(), GODOT_DLOPEN_MODE);
 	ERR_FAIL_NULL_V_MSG(p_library_handle, ERR_CANT_OPEN, vformat("Can't open dynamic library: %s. Error: %s.", p_path, dlerror()));
 
+#if !defined(__ANDROID_API__) && defined(__linux__) || defined(__FreeBSD__) || defined(__NetBSD__)
+	if (path == p_path.get_file()) {
+		const struct link_map *link_map;
+		int err = dlinfo(p_library_handle, RTLD_DI_LINKMAP, &link_map);
+		ERR_FAIL_COND_V_MSG(err != 0, ERR_CANT_RESOLVE, vformat("Failed to resolve dynamic library path: %s. Error: %s.", p_path, dlerror()));
```

Comment:
At this point, we would have succeeded in loaded the library, but only failed in finding the path to the library. If we're going to count this as an overall failure (which this would by returning), I think we should `dlclose()` the library before returning.